tests/test_agent.py

```diff
@@ -179,6 +179,38 @@ def test_agent_get_configuration(self):
             actual_configuration.append("{0} = {1}".format(k, configuration[k]))
         self.assertListEqual(EXPECTED_CONFIGURATION, actual_configuration)
 
+    def test_checks_log_collector_mode(self):
+        # Specify full mode
+        c, f, v, d, cfp, lcm = parse_args(["-collect-logs", "-full"]) # pylint: disable=unused-variable,invalid-name
+        self.assertEqual(c, "collect-logs")
+        self.assertEqual(lcm, True)
+
+        # Defaults to None if mode not specified
+        c, f, v, d, cfp, lcm = parse_args(["-collect-logs"]) # pylint: disable=unused-variable,invalid-name
+        self.assertEqual(c, "collect-logs")
+        self.assertEqual(lcm, False)
+
+    @patch("sys.stderr")
+    @patch("sys.exit", side_effect=Exception)
+    def test_rejects_invalid_log_collector_mode(self, mock_exit, mock_stderr): # pylint: disable=unused-argument
+        try:
+            c, f, v, d, cfp, lcm = parse_args(["-collect-logs", "-mode:invalid"]) # pylint: disable=unused-variable,invalid-name
```

Comment:
Does this test need modification? (`-mode:invalid` is not a valid param anymore)

Reply:
Thanks! I missed it.